python/foglamp/services/core/api/configuration.py

```diff
@@ -596,3 +625,15 @@
     else:
         result = await cf_mgr.get_category_item(category_name, config_item)
         return web.json_response(result)
+
+
+def hide_password(config: dict) -> Dict:
+    new_config = copy.deepcopy(config)
+    try:
+        for k, v in new_config.items():
+            if v['type'] == 'password':
+                v['value'] = "****"
+    except TypeError:
```

Comment:
This is to handle the case of single config item vs category with multiple config items?!

Reply:
yeah, for both cases the instance type is dict but with different structures.